crates/bevy_ui/src/render/mod.rs

```diff
@@ -604,7 +613,7 @@ pub fn queue_uinodes(
                     draw_function: draw_ui_function,
                     pipeline,
                     entity,
-                    sort_key: FloatOrd(batch.z),
+                    sort_key: FloatOrd(0.),
```

Comment:
Not exactly a `wgpu` expert, but from reading [`PhaseItem::sort`](https://dev-docs.bevyengine.org/bevy/render/render_phase/trait.PhaseItem.html#method.sort) I get the impression this is important.

Maybe you thought the batches are already sorted by `stack_index`, so it will render properly using painter's algorithm, no need for extra ordering?

Maybe it helps not run fragment shaders for the same pixel twice if it gets overwritten? But it might not mater if UI rectangles are semi transparent? Idk, you didn't motivate this change.

I think it's safe to replace the `z` from `extracted_uinode.transform.w_axis[2]` by `extracted_ui_node.stack_index`.

If you have good reasons to remove sorting and want to keep it that way, I'd suggest setting `TransparentUi::SortKey` at `crates/bevy_ui/src/render/render_pass.rs` line 96 to an empty tuple:

```rust
    type SortKey = ();
```

Then remove the `sort_key` field of `TransparentUi`.

This makes the lack of sorting explicit.

Reply:
Yep, the commit message about this could have been clearer. 
The `batch.z` field serves no purpose because `extracted_uinode.transform.w_axis[2]` is always equal to 0.

Reply:
Please add these notes as a comment to the code :)